### PR TITLE
fix(grainfmt): Fix comments in records

### LINKED ISSUE #1415: GrainFmt: Grain Formatter Is Duplicating Comments
Similar to #1378 the grain formatter is duplicating my comments again though this time in a record type definition. the first image shows the state before formatting and second image shows the duplicated comments after, every additional format adds another comment. I am running grain v0.5.3 so the fix for #1378 has been implemeneted in the version i am running.
image
image

### PR DESCRIPTION
**Pull request by marcusroberts**
Fixes #1415

### CODE REVIEW

File: compiler/src/formatting/format.re [hunk 1]
```diff
@@ -3644,10 +3644,22 @@ let rec print_data =
       ]);
     };
 
+    let pre_brace_comments = []; // We can't determine from AST if comment comens before or after brace
```
Comment (ospencer) [inline compiler/src/formatting/format.re:3647]:
```suggestion
    let pre_brace_comments = []; // We can't determine from AST if comment comes before or after brace
```

**Review by ospencer — APPROVED** [review]

**Comment by marcusroberts** [comment]
I think I've found another related problem whilst testing #1429, working on it now

**Comment by marcusroberts** [comment]
Found an edge case I need to fix.

**Comment by marcusroberts** [comment]
The edge case is different, I'll raise a new issue

**Review by ospencer — APPROVED** [review]

File: compiler/src/formatting/format.re [hunk 2]
```diff
@@ -3672,22 +3684,11 @@ let rec print_data =
...
-              ~remove_comments=after_angle_comments,
-              comments,
-            );
-
           let param_items =
             item_iterator(
               ~get_loc,
               ~print_item,
-              ~comments=cleaned_comments,
+              ~comments=[],
```
Comment (phated) [inline compiler/src/formatting/format.re:3691]: Does this mean that comments inside parameterized types are no longer support? Do we have a test for something like `type Foo<A /* something */, B> = (A, B)`?

File: compiler/test/formatter_inputs/records.gr [hunk 3]
```diff
@@ -55,8 +74,5 @@ let { // a comment 2
   longlonglongnamenamename3
 } = x
 
-let { // a comment 3
```
Comment (phated) [inline compiler/test/formatter_inputs/records.gr:58]: Why was this removed?

**Review by phated — CHANGES_REQUESTED** [review]
A few questions I need answered.

File: compiler/src/formatting/format.re (hunk 2 above)
Comment (marcusroberts) [reply compiler/src/formatting/format.re:3691]: This is broken, but not by this change, so I'll fix

**Comment by marcusroberts** [comment]
I need to fix #1444  first

File: compiler/src/formatting/format.re (hunk 2 above)
Comment (marcusroberts) [reply compiler/src/formatting/format.re:3813]: Fixed in #1460

File: compiler/test/formatter_inputs/records.gr (hunk 3 above)
Comment (marcusroberts) [reply compiler/test/formatter_inputs/records.gr]: Not sure, it's back now anyway

**Review by phated — COMMENTED** [review]
This LGTM but I want #1460 to land first

**Comment by phated** [comment]
@marcusroberts please resolve the conflicts with `main` now that #1460 is merged 🙇

**Comment by marcusroberts** [comment]
Conflicts removed!

**Review by phated — APPROVED** [review]
